client/index.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+
+
+const e = React.createElement;
+
+// React function component
+const ButtonSearch = props => {
+    const {st, text} = props;
+    return e (
+        'button',
+        { className: 'component', style: { backgroundColor: st ? 'white' : 'pink', color: st ? 'pink' : 'white' } },
+        text
+    )
+}
+
+// React component
+class Search extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { isClicked: false };
+    }
+    render() {
+        return e(
+        'input',
+        { placeholder: this.props.placeholder, className: 'component' , type: 'text', style: { width: '100%', backgroundColor: !this.state.isClicked ? 'black' : 'white', color: !this.state.isClicked ? 'white' : 'black', border: 'none', lineHeight: '30px' }, onClick: () => this.setState({ isClicked: !this.state.isClicked }) },
```

Comment:
1) please do try to avoid inline handlers as they have some impact on performance (each time a new function is being created). introduce a class method:
``` js
handleClick = () => this.setState(({ isClicked } => ({ isClicked: !isClicked })));
```

2) please use `updater` as function in case when you would like to update existing state value: https://reactjs.org/docs/react-component.html#setstate